### DISCUSSION TITLE
Using HA as a bridge to allow Flair to control set point of a unsuppoted thermostat?

### DISCUSSION THREAD

**Discussion by jpettitt** [RobertD502/home-assistant-flair#62] [Q&A] [answered]
I have a Lennox iComfort system, I don't really want to swap out the thermostat becasue it does a pretty decent job of dynamically managing fan speed and current for the AC (see graph below of A/C power usage).  I can control the set point of the Lennox from HA.  If I buy Flair vents and install this integration can I tell Flair to manage the set point, get the target from the integration and use it to adjust the iComfort via HA automation?

<img width="574" alt="image" src="https://github.com/RobertD502/home-assistant-flair/assets/524856/13399266-781b-418c-b033-eb1e695df986">

**Comment by RobertD502** [comment, marked as answer]
Yes, Flair will be able to create a set point which is expose in an entity. You can use this entity to set your thermostat temp.

Is there a particular reason you want to have Flair set your temp? I'd just use the thermostat that you have and manually control the vents (open/close) based on readings from temp sensors placed in the rooms the Flair vents are located.

**Reply by jpettitt** [reply]
The thermostat is in a really dumb place (the kitchen very close to a register, prior homeowner decision). I'm debating moving it which would probably be the better solution.

**Comment by jpettitt** [comment]
Related question - I have zigbee temp sensors in every room , any way I can feed that data to Flair?

**Reply by RobertD502** [reply]
There is no way of writing new sensors to the API, so no you can't feed that data directly to Flair. However, you can use that data to control your Flair vents from within Home Assistant (automate your own logic from the Home Assistant side). Based on what you've shared, there are a few different things that you can do:


## Method 1:
You mentioned that your current thermostat is close to a register. I'm assuming the consequence of this is that the thermostat reaches the set temperature before other rooms in your house. In this case, you'd be better off switching to an Ecobee with remote Ecobee sensors. For example, in your situation, if you set the thermostat to heat to 75 degrees, the thermostat would reach 75 degrees before your bedroom (as an example) and would stop heating with your bedroom never reaching the set point. Putting in smart vents in the rooms that reach the temp first (where the thermostat is located) wouldn't help since those vents won't close until the temp is reached. However, the others rooms will not have reached the set temp yet, but the thermostat will have signaling it to shut off.

If, instead, you employed an Ecobee with remote Ecobee sensors, you can tell the thermostat to use the temp readings from the remote sensors to determine when the set point has been reached and not the thermostat temp reading. This way you can install Flair vents in the rooms that reach the desired temp first and have them close up at that time allowing the air flow to continue to the other rooms. For example, let's say my thermostat is downstairs and I have an office upstairs. The downstairs always reaches the set point before the office. In this example I use a Ecobee thermostat located downstairs, a remote Ecobee sensor in the office, and Flair vents downstairs. I then tell the Ecobee thermostat to use the temp readings from the remote Ecobee sensors in the office instead of the thermostat. I then set the desired temp to 75 degrees. Once the thermostat itself reaches a temperature of 75 degrees, it instructs the downstairs Flair vents to close up - this allows the temp to stay at 75 degrees downstairs without over heating the downstairs and keeps the system running until the upstairs office has reached 75 degrees. If you do use an Ecobee + Sensors, then you don't even have to mess with the logic yourself and can have Flair automatically control the closing and opening of vents for any room that has an ecobee sensor or the ecobee thermostat as a temp reader.

This is not to say that you have to keep on buying Ecobee sensors for rooms that you want to control Flair vents for. For example, I use the above method (Ecobee + Ecobee sensors), but wanted to install another vent in the upstairs bathroom where I don't have any Ecobee sensors, but do have a Zigbee temp/ humidity sensor. In that case, I created a new "room" in the Flair app and called it "Upstairs bathroom" and placed the vent in that Flair room. Since there is no Flair compatible sensors in that Flair room, I handle the opening and closing of the vent manually via Home Assistant. I use the temp readings from the zigbee sensor to control the opening and closing of the upstairs bathroom vent without needing to shell out additional money on another Ecobee sensor or an additional Flair puck:


![image](https://github.com/RobertD502/home-assistant-flair/assets/52541649/56122f89-f55a-49f6-8d15-1f7af0135503)


## Method 2:
Just as I mentioned in the example above where I use a zigbee sensor, you can place your vents in a Flair room that doesn't have a compatible temp sensor in order to control the vent(s) manually. You'd then write out your logic in Home Assistant for when to close/open vents based on the zigbee sensor temp readings. However, this can be problematic - If the rooms close to your thermostat reach the desired set point before some other rooms, you can't just close vents by the thermostat to get those other rooms to heat up/cool down. You'd have to close the vents close to your thermostat prior to the set point being reached (in order to prevent the thermostat from signaling shut off) in order to keep on cooling/heating the other rooms to the desired set point. However, this would put strain on your system since the vents being closed would mean the thermostat will never reach the set point itself and would continue to heat. The only other logic that could be done to handle this is to handle shutting the system off yourself via Home Assistant. For example, let's say the set point that I want to reach is 75. I'd then actually set a set point of 77. Once the downstairs thermostat is reading 75, I instruct the downstairs vents to close to prevent over heating/cooling the downstairs. I then wait until the sensors in the slower to cool/heat rooms reach my desired set point of 75 at which point of time I instruct the thermostat to go into Off mode. This way you're handling shutting the system off since it won't be able to reach the set point of the thermostat with the vents closed.


## Conclusion
As you can see, there are several ways to go about this. However, I'd highly recommend going with the Ecobee route solely because of being able to use remote sensors to dictate when the set point is actually reached. Just as I've done, if you want to place Flair vents in a room that doesn't have a compatible temp sensor (ecobee thermostat, sensor, flair puck), you can utilize your zigbee sensors + manual control of said vents. If you're concerned about local control, Ecobee thermostats can be integrated into Home Assistant via the "Homekit Device" integration which allows for full local control of the thermostat (there is also the Ecobee integration which uses the cloud API).

Hope this helps!


Edit:

Just looked into the iComfort system. Looks like the S40 has remote sensor support like the Ecobee does. Do you happen to own a S40 thermostat? If so, then you can utilize their remote sensors to control what room the thermostat uses to determine if the set point has been reached.
